treeherder/services/intermittents_commenter/commenter.py

```diff
@@ -0,0 +1,284 @@
+        test_run_count = self.get_test_runs(startday, endday)
+
+        template = Template(self.open_file('comment.template', False))
+        components = self.open_file('owner_triage_components.json', True)
+
+        session = self.new_request()
+
+        if self.weekly_mode:
+            top_bugs = [bug[0] for bug in sorted(bug_stats.items(), key=lambda x: x[1]['total'],
+                        reverse=True)][:TOP_BUGS_THRESHOLD]
+
+        for bug_id, counts in bug_stats.iteritems():
+            change_priority = None
+            bug_info = None
+            whiteboard = None
+            priority = 0
+            rank = None
+
+            if self.weekly_mode and bug_id in top_bugs:
+                rank = top_bugs.index(bug_id)+1
+
+            # recommend disabling when more than 150 failures tracked over 21 days
+            if alt_bug_stats[bug_id]['total'] >= DISABLE_THRESHOLD:
+                bug_info = self.fetch_bug_details(session, TRIAGE_PARAMS, bug_id)
+                if bug_info is not None:
+                    whiteboard = bug_info['whiteboard']
+                    if not self.check_whiteboard_status(whiteboard):
+                        priority = 3
+                        whiteboard = self.update_whiteboard(whiteboard, WHITEBOARD_DISABLE_RECOMMENDED)
+
+            if priority == 0 and self.weekly_mode:
+                if counts['total'] >= PRIORITY1_THRESHOLD:
+                    priority = 1
+                elif counts['total'] >= PRIORITY2_THRESHOLD:
+                    priority = 2
+
+            if priority == 2 or not self.weekly_mode:
+                if not bug_info:
+                    bug_info = self.fetch_bug_details(session, TRIAGE_PARAMS, bug_id)
+                    if bug_info is not None:
+                        whiteboard = bug_info['whiteboard']
+
+                        if (([bug_info['product'], bug_info['component']] in components) and
+                            not self.check_whiteboard_status(whiteboard)):
+
+                            if bug_info['priority'] not in ['--', 'P1', 'P2', 'P3']:
+                                change_priority = '--'
+
+                            stockwell_text = re.search(r'\[stockwell (.+?)\]', whiteboard)
+                            if stockwell_text is not None and stockwell_text.group() != WHITEBOARD_NEEDSWORK_OWNER:
+                                whiteboard = self.update_whiteboard(whiteboard, WHITEBOARD_NEEDSWORK_OWNER)
+
+            if self.weekly_mode and (counts['total'] < UNKNOWN_THRESHOLD):
+                if not bug_info:
+                    bug_info = self.fetch_bug_details(session, TRIAGE_PARAMS, bug_id)
+                    if bug_info is not None:
+                        whiteboard = bug_info['whiteboard']
+
+                        stockwell_text = re.search(r'\[stockwell (.+?)\]', whiteboard)
+                        if stockwell_text is not None and stockwell_text.group() == WHITEBOARD_NEEDSWORK:
```

Comment:
What if the existing whiteboard is [stockwell needswork:owner]?

Reply:
Good point - thanks for catching that. I'll modify this section to cover all 'needswork' possibilities.